### ISSUE TITLE
Cannot parse filter (correct LookML)

### ISSUE THREAD

**Issue by mathcoder3141** [joshtemple/lkml#57]
The library is throwing an error when it tries to parse: `filters: [date_day_date: "yesterday"]`

The traceback error is in `parser.py`. Would fix it myself but I don't have enough context on this library.

**Comment by joshtemple** [comment]
Could you please add the error message and the snippet of LookML you're parsing?

**Comment by mathcoder3141** [comment]
Traceback:

```Traceback (most recent call last):
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/IPython/core/interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-9-4b48f32df919>", line 1, in <module>
    lkml.load(lookml)
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/__init__.py", line 33, in load
    result = parser.parse()
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/parser.py", line 153, in parse
    return self.parse_expression()
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/parser.py", line 89, in wrapper
    result = method(self, *args, **kwargs)
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/parser.py", line 250, in parse_expression
    block = self.parse_block()
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/parser.py", line 89, in wrapper
    result = method(self, *args, **kwargs)
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/parser.py", line 322, in parse_block
    expression = self.parse_expression()
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/parser.py", line 89, in wrapper
    result = method(self, *args, **kwargs)
  File "/Users/randall.hall/Library/Python/3.8/lib/python/site-packages/lkml/parser.py", line 266, in parse_expression
    raise SyntaxError(
  File "<string>", line None
SyntaxError: Unable to find a matching expression for ':' on line 5
```

LookML being parsed:

 ```  measure: lagged_prior_arr {
    type: sum
    sql: ${TABLE}.prior_arr  ;;
    filters: [date_day_date: "yesterday"]
    value_format_name: usd_0
  }
```

I tried it both with and without the filters attribute and it works just fine without the attribute.

**Comment by joshtemple** [comment]
Thanks! I'm able to parse the code you pasted, so I need a bit more info from you to debug.

1. What version of lkml are you using?
1. How are you running lkml? From the command line or from Python?
1. If you're running from the command line, could you try again with the debug flag (`--debug`) and post the full logs as an attachment?
1. If you're running from Python, could you please post the code you're using?

**Comment by mathcoder3141** [comment]
Ha! So I just noticed I'm using 0.2.2, which explains quite a lot. (somehow that's the one that got fetched when I installed yesterday) Upgraded to 1.1.0 and it works.

Loosely related: Now that it parsed successfully, I see that `filters` is now `filters_all` and the value is `[{'date_day_date': 'yesterday'}]]`. Is that expected behavior?

**Comment by joshtemple** [comment]
No, that's not expected. That `__all` should be stripped off during parsing. Let me take a look and see if I can get a fix for you later today.

**Comment by joshtemple** [comment]
Hey @mathcoder3141, I clearly need to document this better because I struggled to understand this myself! I believe this actually is expected behavior.

If I'm remembering right, I added `__all` because the key filter(s) can, quite annoyingly, mean three different things in LookML: [1](https://docs.looker.com/reference/field-params/filters), [2](https://docs.looker.com/reference/multilocation-params/filters), [3](https://docs.looker.com/reference/field-params/filter).

When using simple parsing, `lkml` combines multiple blocks with the same key (like `dimension`) into a single, plural key in the parsed dictionary (`{dimensions: ...}`. `__all` is supposed to communicate that all of those `filters` syntaxes get rolled up into a single key.

If you use `lkml.dump`, it should still turn the dictionary into the LookML you would expect.
